docs/web/diff.md

```diff
@@ -0,0 +1,903 @@
+remote-remote, and local-remote or remote-local diffs.
+
+- Compare locally stored analyses `./my_base_results` and locally stored analysis `./my_updated_results`:
+  ```sh
+  CodeChecker cmd diff --basename ./my_base_results --newname ./my_updated_results --new
+  ```
+- Compare remotely stored runs "My Base Results" and "My Updated Results":
+  ```sh
+  CodeChecker cmd diff --basename "My Base Results" --newname "My Updated Results" --new
+  ```
+- Compare remotely stored runs against locally stored analyses:
+  ```sh
+  CodeChecker cmd diff --basename "My Base Results" --newname ./my_updated_results --new
+  CodeChecker cmd diff --basename ./my_base_results --newname "My Updated Results" --new
+  ```
+## Diff through the web GUI
+
+You can also compare runs on the web GUI. On the "Run" page, on the right side of each report, you can see square shaped radio buttons. Clicking on the first one places the run into the "baseline" set, and clicking on the second places it in the "newline" set. After selecting at least one run into each set, you can press the "Diff" button in the upper right corner of the "Run" page.
+
+![image](https://github.com/Szelethus/codechecker/assets/23276031/75490051-86df-419d-998a-ec89942faca1)
+
+Read more [here](../../web/server/vue-cli/src/assets/userguide/userguide.md#compare-runs).
+
+## Diff Example
+
+In this section, we'll showcase an example in between two [runs](../usage.md#definition-of-run). There is an example for diffing tags and timestamps as well [further down](#diff-on-tags-or-timestamps).
+
+### Setup
+Let's assume you have the following C++ code:
+
+```cpp
+// example.cpp
+
+int foo(int z)
+{
+  if (z == 0)
+    return 1 / z; // report by core.DivideZero
+
+  return 0;
+}
+
+int bar(int x)
+{
+  int y;
+  y = x % 2; // report by deadcode.DeadStores
+
+  return x % 2;
+}
+```
+Lets use CodeChecker to analyze this file:
+```sh
+CodeChecker check -o ./test_report_dir -b "g++ example.cpp -c"
+# Alternatively:
+CodeChecker log -o compile_command.json -b "g++ example.cpp -c"
+CodeChecker analyze -o ./test_report_dir compile_command.json
+```
+Using the `parse` command we can see that CodeChecker found and stored
+2 reports in `./test_report_dir` (showing only a part of the output):
+```sh
+CodeChecker parse ./test_report_dir
+```
+```
+----==== Checker Statistics ====----
+--------------------------------------------------
+Checker name        | Severity | Number of reports
+--------------------------------------------------
+core.DivideZero     | HIGH     |                 1
+deadcode.DeadStores | LOW      |                 1
+--------------------------------------------------
+----=================----
+```
+:warning: Note: For the sake of simplictiy, we only enabled the Clang Static Analyzer on these analyses using the `--analyzers=clangsa` flag. We chose omit this flag from our examples. Your actual results may also be different depending on the version of your clang binary.
```

Comment:
The idea was that I'd like to keep these invocations as lean as possible, but still with a managable number of results. So I cheated, used a flag that I later omitted from the examples, which is something I'll likely get away with, but on the off chance somebody does actually try these out, I left this note.

I would strongly not to adjust all invocations to accomodate `--analyzers=clangsa` because it'd be a distractraction.